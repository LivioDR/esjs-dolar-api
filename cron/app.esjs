importar ar desde '@/ar/cron.ar.esjs'
importar cl desde '@/cl/cron.cl.esjs'
importar ve desde '@/ve/cron.ve.esjs'
importar uy desde '@/uy/cron.uy.esjs'
importar mx desde '@/mx/cron.mx.esjs'
importar bo desde '@/bo/cron.bo.esjs'
importar br desde '@/br/cron.br.esjs'
importar { grupo } desde '@/log.esjs'

exportar asincrono funcion iniciar(comando) {
  const log = grupo({ cron: 'app.esjs' })

  log.info('Iniciando cron')

  esperar ar()
  esperar cl()
  esperar ve()
  esperar uy()
  esperar mx()
  esperar bo()
  esperar br()

  log.info('Cron finalizado')
}
