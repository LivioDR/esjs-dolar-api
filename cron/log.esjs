importar pino desde 'pino'

const transports = pino.transport({
  targets: [
    {
      target: 'pino-axiom',
      options: {
        orgId: import.meta.env.VITE_AXIOM_ORG_ID, // Can be found on settings page
        token: import.meta.env.VITE_AXIOM_TOKEN, // Can be created on settings page
        dataset: import.meta.env.VITE_AXIOM_DATASET, // Can be created on settings page
      },
    },
    {
      target: 'pino-pretty',
      options: {
        colorize: verdadero,
        ignore: 'pid,hostname',
      },
    },
  ],
})

const logger = pino(transports)

const traceId =
  Mate.aleatorio().aCadena(36).subcadena(2, 15) +
  Mate.aleatorio().aCadena(36).subcadena(2, 15)

exportar funcion log(message, payload) {
  logger.info(message, {
    traceId,
    ...payload,
  })
}

exportar funcion grupo(grupo) {
  retornar logger.child({
    traceId,
    ...grupo,
  })
}

exportar funcion logError(grupo, error, context) {
  grupo.error({
    msg: 'Error',
    errorMessage: error.message,
    errorStack: error.stack,
    ...context,
  })
}
