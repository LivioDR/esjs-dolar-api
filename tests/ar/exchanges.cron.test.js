import { describe, expect, it } from 'vitest'
import { cronExchanges } from '@/ar/comandos/exchanges.cron.esjs'

describe('exchanges.cron', () => {
  it('cronExchanges', async () => {
    const result = await cronExchanges()

    expect(result).toMatchSnapshot()
  })
})
