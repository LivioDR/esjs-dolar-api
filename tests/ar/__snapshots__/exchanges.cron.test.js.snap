// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`exchanges.cron > cronExchanges 1`] = `
[
  {
    "cantidad": 9,
    "ruta": "/exchanges",
  },
  [
    {
      "cantidad": 5,
      "ruta": "/exchanges/belo/btc",
    },
    {
      "cantidad": 1,
      "ruta": "/exchanges/monedas/ars/btc",
    },
    {
      "cantidad": 1,
      "ruta": "/exchanges/monedas/btc/ars",
    },
    {
      "cantidad": 4,
      "ruta": "/exchanges/belo/eth",
    },
    {
      "cantidad": 1,
      "ruta": "/exchanges/monedas/ars/eth",
    },
    {
      "cantidad": 1,
      "ruta": "/exchanges/monedas/eth/ars",
    },
    {
      "cantidad": 3,
      "ruta": "/exchanges/belo/dai",
    },
    {
      "cantidad": 1,
      "ruta": "/exchanges/monedas/ars/dai",
    },
    {
      "cantidad": 1,
      "ruta": "/exchanges/monedas/dai/ars",
    },
    {
      "cantidad": 6,
      "ruta": "/exchanges/belo/usd",
    },
    {
      "cantidad": 5,
      "ruta": "/exchanges/monedas/ars/usd",
    },
    {
      "cantidad": 5,
      "ruta": "/exchanges/monedas/usd/ars",
    },
    {
      "cantidad": 1,
      "ruta": "/exchanges/monedas/eth/btc",
    },
    {
      "cantidad": 1,
      "ruta": "/exchanges/monedas/btc/eth",
    },
    {
      "cantidad": 1,
      "ruta": "/exchanges/monedas/dai/btc",
    },
    {
      "cantidad": 1,
      "ruta": "/exchanges/monedas/btc/dai",
    },
    {
      "cantidad": 2,
      "ruta": "/exchanges/monedas/usd/btc",
    },
    {
      "cantidad": 2,
      "ruta": "/exchanges/monedas/btc/usd",
    },
    {
      "cantidad": 1,
      "ruta": "/exchanges/monedas/dai/eth",
    },
    {
      "cantidad": 1,
      "ruta": "/exchanges/monedas/eth/dai",
    },
    {
      "cantidad": 2,
      "ruta": "/exchanges/monedas/usd/eth",
    },
    {
      "cantidad": 2,
      "ruta": "/exchanges/monedas/eth/usd",
    },
    {
      "cantidad": 2,
      "ruta": "/exchanges/monedas/usd/dai",
    },
    {
      "cantidad": 2,
      "ruta": "/exchanges/monedas/dai/usd",
    },
    {
      "cantidad": 6,
      "ruta": "/exchanges/monedas/usd/usd",
    },
    {
      "cantidad": 2,
      "ruta": "/exchanges/belo/brl",
    },
    {
      "cantidad": 7,
      "ruta": "/exchanges/monedas/usd/brl",
    },
    {
      "cantidad": 7,
      "ruta": "/exchanges/monedas/brl/usd",
    },
    {
      "cantidad": 7,
      "ruta": "/exchanges/monedas/ars/brl",
    },
    {
      "cantidad": 7,
      "ruta": "/exchanges/monedas/brl/ars",
    },
    {
      "cantidad": 2,
      "ruta": "/exchanges/fiwind/brl",
    },
    {
      "cantidad": 1,
      "ruta": "/exchanges/cocos/usd",
    },
    {
      "cantidad": 1,
      "ruta": "/exchanges/cocos/brl",
    },
    {
      "cantidad": 1,
      "ruta": "/exchanges/dolarapp/usd",
    },
    {
      "cantidad": 1,
      "ruta": "/exchanges/takenos/usd",
    },
    {
      "cantidad": 2,
      "ruta": "/exchanges/satoshitango/usd",
    },
    {
      "cantidad": 1,
      "ruta": "/exchanges/satoshitango/brl",
    },
    {
      "cantidad": 1,
      "ruta": "/exchanges/prex/brl",
    },
    {
      "cantidad": 1,
      "ruta": "/exchanges/prex/usd",
    },
    {
      "cantidad": 2,
      "ruta": "/exchanges/lemon/brl",
    },
    {
      "cantidad": 1,
      "ruta": "/exchanges/monedas/usdt/brl",
    },
    {
      "cantidad": 1,
      "ruta": "/exchanges/monedas/brl/usdt",
    },
    {
      "cantidad": 1,
      "ruta": "/exchanges/plus/usd",
    },
    {
      "cantidad": 1,
      "ruta": "/exchanges/plus/brl",
    },
  ],
]
`;
